lib/realtime/tenants/connect.ex

```diff
@@ -77,6 +97,41 @@ defmodule Realtime.Tenants.Connect do
     end
   end
 
+  @impl GenServer
+  def handle_continue(
+        :setup_connected_users,
+        %{
+          check_connected_user_interval: check_connected_user_interval,
+          connected_users_bucket: connected_users_bucket
+        } = state
+      ) do
+    send_connected_user_check_message(connected_users_bucket, check_connected_user_interval)
+    {:noreply, state}
+  end
+
+  @impl GenServer
+  def handle_info(
+        :check_connected_users,
+        %{
+          tenant_id: tenant_id,
+          check_connected_user_interval: check_connected_user_interval,
+          connected_users_bucket: connected_users_bucket
+        } = state
+      ) do
+    connected_users_bucket =
+      tenant_id
+      |> update_connected_users_bucket(connected_users_bucket)
+      |> send_connected_user_check_message(check_connected_user_interval)
+
+    {:noreply, %{state | connected_users_bucket: connected_users_bucket}}
+  end
+
+  def handle_info(:shutdown, %{db_conn_pid: db_conn_pid} = state) do
+    Logger.info("Tenant has no connected users, database connection will be terminated")
+    :ok = GenServer.stop(db_conn_pid)
```

Comment:
What happens after this? Is the process going to keep checking and trying to kill a db_conn_pid that doesn't exist?

Reply:
Since we're monitoring the reference of this connection it will be caught by the handle info for that effect and kill itself

Reply:
Roger. Okay LGTM.